### PR TITLE
Avoid the same session id for openai endpoint

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: lmdeploy/serve/openai/api_server.py [hunk 1]
```diff
@@ -416,11 +416,17 @@ async def chat_completions_v1(request: ChatCompletionRequest,
...
+        VariableInterface.session_id += 1
+        request.session_id = VariableInterface.session_id
     error_check_ret = await check_request(request)
     if error_check_ret is not None:
         return error_check_ret
+    if VariableInterface.async_engine.id2step.get(str(request.session_id),
+                                                  0) != 0:
+        return create_error_response(
+            HTTPStatus.BAD_REQUEST,
+            f'The session_id `{request.session_id}` is occupied.')
```
Comment (zhyncs) [inline lmdeploy/serve/openai/api_server.py:429]: Hi @AllentDan Under what circumstances will multiple different requests use the same session id?

Comment (AllentDan) [reply lmdeploy/serve/openai/api_server.py:429]: they use both `interactive` and `completions` endpoints.

File: lmdeploy/serve/openai/api_server.py [hunk 2]
```diff
@@ -416,11 +416,17 @@ async def chat_completions_v1(request: ChatCompletionRequest,
...
     """
-    VariableInterface.session_id += 1
-    request.session_id = VariableInterface.session_id
+    if request.session_id == -1:
+        VariableInterface.session_id += 1
+        request.session_id = VariableInterface.session_id
     error_check_ret = await check_request(request)
     if error_check_ret is not None:
         return error_check_ret
+    if VariableInterface.async_engine.id2step.get(str(request.session_id),
```
Comment (zhyncs) [inline lmdeploy/serve/openai/api_server.py:425]: nit: Use `in` to determine whether the key exists in the `dict`. Will the readability be better? Is there a difference in performance compared to `get`? What is your suggestion? @AllentDan

Comment (AllentDan) [reply lmdeploy/serve/openai/api_server.py:425]: I just set the step to 0 instead of popping it out from `id2step` in `generate` func

Comment (zhyncs) [reply lmdeploy/serve/openai/api_server.py:425]: ok

**Review by zhyncs — APPROVED** [review]

File: lmdeploy/serve/openai/api_server.py (hunk 2 above)
Comment (lvhan028) [inline lmdeploy/serve/openai/api_server.py:425]: Can we move it to `check_request` function?

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**